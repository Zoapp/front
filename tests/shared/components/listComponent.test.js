import React from 'react';
import renderer from 'react-test-renderer';
import ListComponent from '@shared/components/listComponent';

describe('ListComponent', () => {
  const newItem = ({ id, name, icon = null }) => {
    return {
      id: id || name,
      name,
      icon,
    };
  };

  it('renders correctly', () => {
    const items = [];
    const component = renderer.create(
      <ListComponent
        items={items}
        selectedItem={0}
        onSelect={jest.fn()}
        className="some-class"
        style={{}}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders items', () => {
    const items = [
      newItem({ name: 'item 1' }),
      newItem({ name: 'item 2' }),
      newItem({ name: 'item 3' }),
    ];
    const component = renderer.create(
      <ListComponent
        items={items}
        selectedItem={0}
        onSelect={jest.fn()}
        className="some-class"
        style={{}}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders items with icons', () => {
    const items = [
      newItem({ name: 'item 1', icon: 'some-icon' }),
    ];
    const component = renderer.create(
      <ListComponent
        items={items}
        selectedItem={0}
        onSelect={jest.fn()}
        className="some-class"
        style={{}}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('marks an item as selected', () => {
    const items = [
      newItem({ name: 'item 1' }),
      newItem({ name: 'item 2' }),
      newItem({ name: 'item 3' }),
    ];
    const component = renderer.create(
      <ListComponent
        items={items}
        selectedItem={1}
        onSelect={jest.fn()}
        className="some-class"
        style={{}}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders PNG icon', () => {
    const items = [
      newItem({ name: 'item 1', icon: 'some-icon.png' }),
    ];
    const component = renderer.create(
      <ListComponent
        items={items}
        selectedItem={0}
        onSelect={jest.fn()}
        className="some-class"
        style={{}}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders SVG icon', () => {
    const items = [
      newItem({ name: 'item 1', icon: 'some-icon.svg' }),
    ];
    const component = renderer.create(
      <ListComponent
        items={items}
        selectedItem={0}
        onSelect={jest.fn()}
        className="some-class"
        style={{}}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
