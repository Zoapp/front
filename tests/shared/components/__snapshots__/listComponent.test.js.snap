// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent marks an item as selected 1`] = `
<ul
  className="mdl-list some-class"
  style={Object {}}
>
  <li
    className="mdl-list__item selectableListItem"
    onClick={[Function]}
  >
    <span
      className="mdl-list__item-primary-content"
    >
      <span>
        item 1
      </span>
    </span>
  </li>
  <li
    className="mdl-list__item selectedListItem"
    onClick={[Function]}
  >
    <span
      className="mdl-list__item-primary-content"
    >
      <span>
        item 2
      </span>
    </span>
  </li>
  <li
    className="mdl-list__item selectableListItem"
    onClick={[Function]}
  >
    <span
      className="mdl-list__item-primary-content"
    >
      <span>
        item 3
      </span>
    </span>
  </li>
</ul>
`;

exports[`ListComponent renders PNG icon 1`] = `
<ul
  className="mdl-list some-class"
  style={Object {}}
>
  <li
    className="mdl-list__item selectedListItem"
    onClick={[Function]}
  >
    <span
      className="mdl-list__item-primary-content"
    >
      <div
        className="mdl-list__item-avatar"
        style={
          Object {
            "backgroundColor": "gray",
            "padding": "8px",
          }
        }
      >
        <img
          alt="item 1"
          src="some-icon.png"
          style={
            Object {
              "margin": "-7px",
              "verticalAlign": "top",
              "width": "38px",
            }
          }
        />
      </div>
      <span>
        item 1
      </span>
    </span>
  </li>
</ul>
`;

exports[`ListComponent renders SVG icon 1`] = `
<ul
  className="mdl-list some-class"
  style={Object {}}
>
  <li
    className="mdl-list__item selectedListItem"
    onClick={[Function]}
  >
    <span
      className="mdl-list__item-primary-content"
    >
      <div
        className="mdl-list__item-avatar"
        style={
          Object {
            "backgroundColor": "gray",
            "padding": "8px",
          }
        }
      >
        <img
          alt="item 1"
          src="some-icon.svg"
          style={
            Object {
              "verticalAlign": "top",
            }
          }
        />
      </div>
      <span>
        item 1
      </span>
    </span>
  </li>
</ul>
`;

exports[`ListComponent renders correctly 1`] = `
<ul
  className="mdl-list some-class"
  style={Object {}}
/>
`;

exports[`ListComponent renders items 1`] = `
<ul
  className="mdl-list some-class"
  style={Object {}}
>
  <li
    className="mdl-list__item selectedListItem"
    onClick={[Function]}
  >
    <span
      className="mdl-list__item-primary-content"
    >
      <span>
        item 1
      </span>
    </span>
  </li>
  <li
    className="mdl-list__item selectableListItem"
    onClick={[Function]}
  >
    <span
      className="mdl-list__item-primary-content"
    >
      <span>
        item 2
      </span>
    </span>
  </li>
  <li
    className="mdl-list__item selectableListItem"
    onClick={[Function]}
  >
    <span
      className="mdl-list__item-primary-content"
    >
      <span>
        item 3
      </span>
    </span>
  </li>
</ul>
`;

exports[`ListComponent renders items with icons 1`] = `
<ul
  className="mdl-list some-class"
  style={Object {}}
>
  <li
    className="mdl-list__item selectedListItem"
    onClick={[Function]}
  >
    <span
      className="mdl-list__item-primary-content"
    >
      <div
        className="mdl-list__item-avatar"
        style={
          Object {
            "backgroundColor": "gray",
            "padding": "8px",
          }
        }
      >
        <i
          className="material-icons"
          style={
            Object {
              "verticalAlign": "top",
            }
          }
        >
          some-icon
        </i>
      </div>
      <span>
        item 1
      </span>
    </span>
  </li>
</ul>
`;
